### PR TITLE
raftstore: consider wake-up msg when checking stale msg in peer fsm

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: Liqi Geng <gengliqiii@gmail.com>

### What problem does this PR solve?

Problem Summary:
In https://github.com/tikv/tikv/pull/7416, we check wake-up msg in store fsm, this logic will be triggered only when the asked peer has already destroyed. But if the asked peer exists, it will ignore the wake-up msg and the problem is similar to https://github.com/tikv/tikv/pull/7416.
Although the learner who sends wake-up msg can be destroyed after querying to PD in this case, it needs several minutes which is too slow.

### What is changed and how it works?

What's Changed:
Do not ignore the wake-up msg in peer fsm.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test

Side effects

No

### Release note 
Using the same release note as https://github.com/tikv/tikv/pull/7416

### CODE REVIEW

File: tests/integrations/raftstore/test_merge.rs [hunk 1]
```diff
@@ -1037,3 +1036,50 @@ fn test_merge_isloated_not_in_merge_learner() {
     // We can see if the old peer 2 is destroyed
     must_get_equal(&cluster.get_engine(2), b"k123", b"v123");
 }
+
+// Test if a learner can be destroyed properly when it's isloated and removed by conf change
+// before another region merge to its region
+#[test]
+fn test_merge_isloated_stale_learner() {
```
Comment (BusyJay) [inline tests/integrations/raftstore/test_merge.rs:1043]: Merge doesn't seem to be necessary in this case.

Comment (gengliqi) [reply tests/integrations/raftstore/test_merge.rs:1043]: Because it's a learner. Merge and then split are used to trigger a learner to send a wake-up msg.

**Pull request merged**